### PR TITLE
Deprecating groupCol and blockCol 

### PR DESCRIPTION
**Pull request by asyakhl**
Replacing  groupCol and blockCol  with classCol and subclassCol, and other group and block variable names. Also updated man files.

### CODE REVIEW

**Comment by shbrief** [comment]
This pull request seems to include only the new argument names but doesn't have a deprecation process. I found this guide (https://contributions.bioconductor.org/deprecation.html) on function and dataset-level deprecations. @LiNk-NY Do you know whether there is a separate instruction on argument-level deprecations?

File: vignettes/lefser.Rmd [hunk 1]
```diff
@@ -31,7 +31,7 @@ The original software utilizes standard statistical significance tests along
 with supplementary tests that incorporate biological consistency and the 
 relevance of effects to identity the features (e.g., organisms, clades, OTU, 
 genes, or functions) that are most likely to account for differences between 
-the two sample groups of interest, referred as ‘classes’. While *LEfSe* is 
+the two sample classes of interest, referred as ‘classes’. While *LEfSe* is
```
Comment (LiNk-NY) [inline vignettes/lefser.Rmd:34]: This is redundant

File: R/lefser.R [hunk 2]
```diff
@@ -319,18 +317,18 @@ lefser <-
...
     ## Add attributes with argument values
     ## This is used for plottin functions
     attr(res_scores, "inputSE") <- relab
     attr(res_scores, "kth") <-  kruskal.threshold
     attr(res_scores, "wth") <- wilcox.threshold
     attr(res_scores, "ldath") <- lda.threshold
-    attr(res_scores, "grp") <- groupCol
-    attr(res_scores, "blk") <- blockCol
+    attr(res_scores, "grp") <- classCol
+    attr(res_scores, "blk") <- subclassCol
```
Comment (LiNk-NY) [inline R/lefser.R:329]: These attribute names should also change

File: R/lefser.R [hunk 3]
```diff
@@ -213,21 +211,21 @@ filterKruskal <- function(relab, group, p.value, method = method) {
...
 #' @export
 lefser <-
   function(relab,
            kruskal.threshold = 0.05,
            wilcox.threshold = 0.05,
            lda.threshold = 2.0,
-           groupCol = "GROUP",
-           blockCol = NULL,
+           classCol = "CLASS",
+           subclassCol = NULL,
```
Comment (LiNk-NY) [inline R/lefser.R:228]: The lefser function shoul dkeep the `groupCol` and `blockCol` in the function signature. They should go after the `...` similarly to `expr`.

**Review by LiNk-NY — CHANGES_REQUESTED** [review]
- [x] Update the main `lefser` function to deprecate the arguments similar to the `expr` argument. 
- [x] Remove redundant text from documentation

**Comment by LiNk-NY** [comment]
ping @asyakhl

File: R/lefser.R [hunk 4]
```diff
@@ -325,8 +327,8 @@ lefser <-
     attr(res_scores, "kth") <-  kruskal.threshold
     attr(res_scores, "wth") <- wilcox.threshold
     attr(res_scores, "ldath") <- lda.threshold
-    attr(res_scores, "grp") <- classCol
-    attr(res_scores, "blk") <- subclassCol
+    attr(res_scores, "class") <- classCol
+    attr(res_scores, "subclass") <- subclassCol
```
Comment (LiNk-NY) [inline R/lefser.R:331]: @asyakhl Were these updated in `lefsePlotFeat.R` ?

**Review by LiNk-NY — CHANGES_REQUESTED** [review]
Please update `lefsePlotFeat.R` to changes

**Review by LiNk-NY — APPROVED** [review]
Thanks Asya! @asyakhl 
Other than the `man/lefserPlotFeat.Rd`, everything looks in order :) 
I think the merge commits got a little bit messy. 
For next time, use `rebase` often to pull changes from `devel` and to keep a linear history.

File: man/lefserPlotFeat.Rd
[the remark this answers is not in the document]
Comment (shbrief) [reply man/lefserPlotFeat.Rd:1]: Oh... sorry, I was cleaning up to reflect the contributions in the repo and accidentally deleted that line. I'll put it back.

File: DESCRIPTION [hunk 5]
```diff
@@ -5,14 +5,13 @@ Description: lefser is the R implementation of the popular microbiome
...
-Version: 1.15.8
-Date: 2024-09-06
+Version: 1.15.9
+Date: 2024-09-19
 Authors@R: c(
   person("Sehyun", "Oh", , "shbrief@gmail.com", c("cre", "ctb"), c(ORCID = "0000-0002-9490-3061")),
   person("Asya", "Khleborodova", , "asya.bioconductor@gmail.com", "aut"),
-  person("Marcel", "Ramos", , "marcel.ramos@roswellpark.org", "ctb", c(ORCID = "0000-0002-3242-0582")),
   person("Samuel", "Gamboa-Tuz", , "Samuel.Gamboa.Tuz@gmail.com", "ctb"),
-  person("Ludwig", "Geistlinger", , "Ludwig.Geistlinger@sph.cuny.edu", "ctb", c(ORCID = "0000-0002-2495-5464")),
```
[the remark this answers is not in the document]
Comment (shbrief) [reply DESCRIPTION]: Sorry... I was cleaning up to reflect the contributions to the repo and accidentally deleted the line. I'll put it back.

File: man/lefserPlotFeat.Rd
[the remark this answers is not in the document]
Comment (shbrief) [reply man/lefserPlotFeat.Rd:1]: Maybe incorporated during `rebase`?

File: R/lefserPlotFeat.R [hunk 6]
```diff
@@ -29,34 +29,34 @@
...
 #' # (2) Using classes and sub-classes
-#' res_block <- lefser(zeller14tn_ra,
-#'                     groupCol = "study_condition",
-#'                     blockCol = "age_category")
-#' plot_group <- lefserPlotFeat(res_group, res_group$features[[1]])
-#' plot_block <- lefserPlotFeat(res_block, res_block$features[[2]])
+#' res_subclass <- lefser(zeller14tn_ra,
+#'                     classCol = "study_condition",
+#'                     subclassCol = "age_category")
+#' plot_class <- lefsePlotFeat(res_class, res_class$features[[1]])
```
Comment (shbrief) [inline R/lefserPlotFeat.R:38]: @asyakhl The correct function name is `lefserPlotFeat`, not `lefsePlotFeat`

**Comment by LiNk-NY** [comment]
Thanks Asya! @asyakhl 
I squashed the commits and added a couple of changes at #65. I keep your authorship with `Co-authored-by: <...> `tag.

**Pull request closed without merging**